### PR TITLE
OCPQE-13266 : Add baremetal-upi-ovn-ipv4-disconnected CI profile

### PR DESCRIPTION
**Pull request by jadhaj**
The PR implements baremetal-upi-ovn-ipv4-disconnected profile by adding a new workflow that includes needed steps to make UPI deployment disconnected and voa proxy

### CODE REVIEW

**Comment by jadhaj** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-upi-ovn-ipv4-disconnected-proxy-p1-f14

**Comment by jadhaj** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-upi-ovn-ipv4-disconnected-proxy-p1-f14

**Comment by jadhaj** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-upi-ovn-ipv4-disconnected-proxy-p1-f14

**Comment by jadhaj** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-upi-ovn-ipv4-disconnected-proxy-p1-f14

**Comment by jadhaj** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-baremetal-upi-ovn-ipv4-disconnected-proxy-p1-f14

**Comment by jadhaj** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-baremetal-upi-ovn-ipv4-disconnected-proxy-p1-f14

File: ci-operator/step-registry/baremetal/lab/upi/install/disconnected/proxy/baremetal-lab-upi-install-disconnected-proxy-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      - chain: baremetal-lab-upi-conf-disconnected-proxy
+      - chain: baremetal-lab-upi-install
+    post:
+      - chain: baremetal-lab-post
+    env:
+      DISCONNECTED: "true"
+      CLUSTER_WIDE_PROXY: "true"
+  documentation: |-
+    The baremetal-lab-upi workflow provides pre- and post- steps that provision and
+    deprovision an UPI OpenShift cluster with a default configuration on a Baremetal lab,
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/upi/install/disconnected/proxy/baremetal-lab-upi-install-disconnected-proxy-workflow.yaml:14]: can you fix the documentation @jadhaj ?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 2]
```diff
@@ -2017,6 +2017,21 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-automated-release
     workflow: cucushift-installer-rehearse-vsphere-upi-zones
+- as: baremetal-upi-ovn-ipv4-disconnected-proxy-p1-f14
```
Comment (aleskandro) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:2020]: can we move this to the multi file and set it for 4.16? We could have a multi-payload-based job on amd64 only for this configuration.

**Comment by aleskandro** [comment]
Can you fix the history of commits?

**Comment by jadhaj** [comment]
Done

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-upi-ovn-ipv4-disconnected-proxy-p1-f14

**Comment by jadhaj** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-baremetal-upi-ovn-ipv4-disconnected-proxy-p1-f14

**Comment by jadhaj** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-baremetal-upi-ovn-ipv4-disconnected-proxy-p1-f14

**Comment by jadhaj** [comment]
@aleskandro Now that profile move, I think I should add:
dependencies:
  MULTI_RELEASE_IMAGE: release:multi-latest
  OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:multi-latest

Also the profile I added is for amd. Should I add one for arm?

**Comment by jadhaj** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-upi-ovn-ipv4-disconnected-proxy-arm-p1-f14

**Comment by jadhaj** [comment]
deployment passed after last changes:
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/50417/rehearse-50417-periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-upi-ovn-ipv4-disconnected-proxy-arm-p1-f14/1777653095312920576

**Comment by jadhaj** [comment]
@aleskandro History fixed.
I think PR is ready now

**Comment by jadhaj** [comment]
I also added a missing profile for arm. when I moved to 4.16 multi, I missed it

**Comment by aleskandro** [comment]
Hey, Jad we don't need  to test the same scenario... Can we set this only in arm And give a final shot for testing?

**Comment by jadhaj** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-upi-ovn-ipv4-disconnected-proxy-arm-p1-f14

**Comment by jadhaj** [comment]
Done

**Review by aleskandro — APPROVED** [review]

**Comment by jadhaj** [comment]
@jhou1 needs your approval. Thanks

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml [hunk 3]
```diff
@@ -2032,6 +2032,23 @@ tests:
       WORKFLOW_TYPE: cucushift
       YQ_VERSION: 4.30.5
     workflow: openshift-e2e-ibmcloud-heterogeneous-test-qe
+- as: baremetal-upi-ovn-ipv4-disconnected-proxy-arm-p1-f14
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml:2035]: `-pN (n=1,2,3,...)` is no longer needed.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml [hunk 4]
```diff
@@ -2032,6 +2032,23 @@ tests:
...
+  cluster: build05
+  cron: 30 21 1,14 * *
+  steps:
+    cluster_profile: equinix-ocp-metal-qe
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:multi-latest
+    env:
+      AUX_HOST: openshift-qe-metal-ci.arm.eng.rdu2.redhat.com
+      E2E_RUN_TAGS: '@baremetal-upi and @amd64 and @disconnected and @network-ovnkubernetes
+        and not @fips'
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml:2045]: For `E2E_RUN_TAGS`, only need `@disconnected` here. The other ones can be dynamically detected.

File: ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13-periodics.yaml [hunk 5]
```diff
@@ -22226,6 +22226,17 @@ periodics:
     job-release: "4.13"
     pj-rehearse.openshift.io/can-be-rehearsed: "true"
   name: periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-openshift-logging-5.7-gcp-ipi-proxy-etcd-encryption-f7
+  reporter_config:
+    slack:
+      channel: '#aos-qe-logging'
```
Comment (liangxia) [inline ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13-periodics.yaml:22231]: Are those slack notification changes for logging expected in this PR ?

Comment (jadhaj) [reply ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13-periodics.yaml:22231]: I believe no. Removing

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml (hunk 3 above)
Comment (jadhaj) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml:2035]: Not sure I got what need to fix here

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml:2035]: We no long need `-p1` in the test name.

File: ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13-periodics.yaml [hunk 6]
```diff
@@ -33680,6 +33680,17 @@ periodics:
     job-release: "4.13"
     pj-rehearse.openshift.io/can-be-rehearsed: "true"
   name: periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-openshift-logging-5.7-azure-upi-f28-disruptive
+  reporter_config:
+    slack:
+      channel: '#aos-qe-logging'
```
Comment (liangxia) [inline ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13-periodics.yaml:33685]: Another slack notification not related to changes in this PR.

**Comment by liangxia** [comment]
A required job failed. `ci/prow/generated-config — Job failed.`
Could you run `make update ( or make jobs)` to fix it ?

**Comment by jadhaj** [comment]
make update done

**Pull request merged**